File: kata-setup-typescript/src/BackstageItem.ts
import { Item, ItemType } from "./Item.js"

export class Backstage extends Item {
  constructor(name: string, sellIn: number, quality: number) {
    super(name, sellIn, quality, ItemType.Backstage)
  }

  updateQuality(): void {
    // Legendary items don't change
    if (this.sellIn < 0) {
      this.resetQuality()
    } else if (this.sellIn <= 5) {
      this.increaseQuality(3)
    } else if (this.sellIn <= 10) {
      this.increaseQuality(2)
    } else {
      this.increaseQuality(1)
    }
  }
}

File: kata-setup-java/src/main/java/org/example/BackstageItem.java
package org.example;

public class BackstageItem extends Item{

  public BackstageItem(int sellIn, int quality) {
    super("Backstage passes to a TAFKAL80ETC concert", sellIn, quality);
  }

  @Override
  public void updateQuality() {
    if (sellIn == 0) {
      quality = 0;
    } else if (sellIn < 5) {
      quality = quality + 3;
    } else if (sellIn < 10) {
      quality = quality + 2;
    }
  }
}
